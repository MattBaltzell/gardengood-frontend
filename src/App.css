.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: 2rem;
  background-color: #282c34;
  color: rgba(255, 255, 255, 0.6);
}

.container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  gap: 2rem;
  width: 100%;
  max-width: 1200px;
}

.Plant {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: left;
  justify-content: flex-start;
  width: 100%;
  text-align: center;
  /* background-color: #3d4553; */
  gap: 1rem;
  border-radius: 3px;
  /* box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2); */

  border-radius: 50%;
}

.Plant h3 {
  font-size: 1rem;
  transition: all 0.2s ease-in-out;
}

.Plant-img {
  object-fit: cover;
  width: 100%;
  border-radius: 50%;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3);
  position: relative;
  transform: translateY(0);
  transition: all 0.2s ease-in-out;
}

.Plant-img:hover {
  filter: brightness(1.1);
  transform: translateY(-0.5rem) scale(1.05);
  box-shadow: 0 8px 16px rgba(0, 0, 0, 0.4);
}
.Plant:hover h3 {
  color: #fff;
  transform: translateY(-0.3rem) scale(1.04);
}
