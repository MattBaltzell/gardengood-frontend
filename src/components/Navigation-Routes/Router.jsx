import React, { useContext } from "react";
import { Routes, Route } from "react-router-dom";
import PlantSearch from "../Plant/PlantSearch";
import Plant from "../Plant/Plant";
import Home from "../../Home/Home";
import SignupForm from "../../auth/SignupForm";
import LoginForm from "../../auth/LoginForm";
import UserContext from "../../auth/UserContext";
import PrivateRoutes from "./PrivateRoutes";

const Router = ({ signup, login, handleIsLoading, isLoading, toast }) => {
  const currUser = useContext(UserContext);

  return (
    <div className="container">
      <Routes>
        <Route path="/" element={<Home />} />

        <Route element={<PrivateRoutes />}>
          <Route
            element={
              <PlantSearch
                isLoading={isLoading}
                handleIsLoading={handleIsLoading}
              />
            }
            path="/plants"
            exact
          />
          <Route element={<Plant />} path="/plants/:id" />
        </Route>

        <Route
          path="/signup"
          element={<SignupForm signup={signup} toast={toast} />}
        />
        <Route
          path="/login"
          element={<LoginForm login={login} toast={toast} />}
        />

        <Route
          path="*"
          element={
            <>
              <h1>404: PAGE NOT FOUND</h1>
            </>
          }
        />
      </Routes>
    </div>
  );
};

export default Router;
