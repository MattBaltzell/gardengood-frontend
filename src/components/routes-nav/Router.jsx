import React from "react";
import { Routes, Route } from "react-router-dom";
import PlantSearch from "../plant/PlantSearch";
import Plant from "../plant/Plant";
import User from "../profile/User";
import Home from "../home/Home";
import SignupForm from "../auth/SignupForm";
import LoginForm from "../auth/LoginForm";
import UpdateUserForm from "../profile/UpdateUserForm";

import PrivateRoutes from "./PrivateRoutes";

const Router = ({
  signup,
  login,
  update,
  handleIsLoading,
  isLoading,
  toast,
}) => {
  return (
    <div className="container">
      <Routes>
        <Route path="/" element={<Home />} />

        <Route element={<PrivateRoutes />}>
          <Route
            element={
              <PlantSearch
                isLoading={isLoading}
                handleIsLoading={handleIsLoading}
              />
            }
            path="/plants"
            exact
          />
          <Route element={<Plant />} path="/plants/:id" />
          <Route
            element={<UpdateUserForm update={update} />}
            path="/users/:username/update"
            exact
          />
          <Route element={<User />} path="/users/:username" exact />
        </Route>

        <Route
          path="/signup"
          element={<SignupForm signup={signup} toast={toast} />}
        />
        <Route
          path="/login"
          element={<LoginForm login={login} toast={toast} />}
        />

        <Route
          path="*"
          element={
            <>
              <h1>404: PAGE NOT FOUND</h1>
            </>
          }
        />
      </Routes>
    </div>
  );
};

export default Router;
