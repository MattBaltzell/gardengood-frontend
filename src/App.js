import { useState, useEffect } from "react";
import "./App.css";
import axios from "axios";

import PlantGrid from "./components/Plant/PlantGrid";
function App() {
  const [plants, setPlants] = useState([]);

  useEffect(() => {
    async function getPlants() {
      try {
        const res = await axios({
          url: "http://localhost:3001/plants",
          method: "GET",
          headers: {
            Authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3RhZG1pbiIsImlzQWRtaW4iOmZhbHNlLCJpYXQiOjE2Nzc2NDA3Mzl9.HoNDrjYXseNCN6CIKkCB9FqT6ecJumAVNE6OeTg1WLk`,
          },
        });
        const plants = res.data.plants;

        setPlants(plants);
      } catch (error) {
        console.error(error);
      }
    }
    getPlants();
  }, []);
  return (
    <div className="App">
      <div className="container">
        <PlantGrid plants={plants} />
      </div>
    </div>
  );
}

export default App;
