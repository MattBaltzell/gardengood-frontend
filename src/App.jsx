import React, { useState, useEffect } from "react";
import "./App.css";
import Router from "./components/Navigation-Routes/Router";
import Navbar from "./components/Navigation-Routes/Navbar";
import GardenGoodApi from "./api/api";
import UserContext from "./auth/UserContext";
import jwt_decode from "jwt-decode";

function App() {
  const [currUser, setCurrUser] = useState(null);
  const [token, setToken] = useState(localStorage.getItem("gardengood-token"));
  const [userWasUpdated, setUserWasUpdated] = useState(false);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getCurrUser = async () => {
      if (token) {
        try {
          GardenGoodApi.token = token;
          let { username } = jwt_decode(token);
          const user = await GardenGoodApi.getCurrentUser(username);
          setCurrUser(user);
          localStorage.setItem("gardengood-token", token);
        } catch (error) {
          console.error(error);
        }
      }
    };
    getCurrUser();

    setUserWasUpdated(false);
  }, [token, userWasUpdated]);

  const handleIsLoading = (bool) => {
    setIsLoading(bool);
  };

  const login = async ({ username, password }) => {
    try {
      const token = await GardenGoodApi.login({ username, password });
      setToken(token);
    } catch (error) {
      return { message: error };
    }
  };

  const signup = async ({
    username,
    password,
    firstName,
    lastName,
    email,
    zipCode,
  }) => {
    try {
      const token = await GardenGoodApi.signup({
        username,
        password,
        firstName,
        lastName,
        email,
        zipCode,
      });
      setToken(token);
    } catch (error) {
      return { message: error };
    }
  };

  const logout = async () => {
    try {
      setToken(null);
      setCurrUser(null);
      GardenGoodApi.token = null;
      localStorage.setItem("gardengood-token", "");
    } catch (error) {
      return { message: error };
    }
  };

  return (
    <div className="App">
      <UserContext.Provider value={currUser}>
        <Navbar logout={logout} />
        <Router
          login={login}
          signup={signup}
          logout={logout}
          handleIsLoading={handleIsLoading}
          isLoading={isLoading}
        />
      </UserContext.Provider>
    </div>
  );
}

export default App;
