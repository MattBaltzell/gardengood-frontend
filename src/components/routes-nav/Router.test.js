import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Router from "./Router";
import App from "../../App";
import UserContext from "../auth/UserContext";

const user = {
  username: "testuser",
  first_name: "firstname",
  last_name: "lastname",
  email: "test@test.com",
};

describe("Router component tests", function () {
  it("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Router />
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={user}>
          <Router />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot when logged out", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={null}>
          <Router />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("redirects to login anon", function () {
    render(
      <MemoryRouter initialEntries={["/plants"]}>
        <UserContext.Provider value={null}>
          <App />
        </UserContext.Provider>
      </MemoryRouter>
    );
    expect(screen.getByText("No Account?")).toBeInTheDocument();
  });
});
