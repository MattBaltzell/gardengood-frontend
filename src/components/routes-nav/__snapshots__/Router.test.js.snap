// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="Home"
    >
      <div
        class="Home__bg"
      />
      <img
        alt="GardenGood"
        class="Home__logo"
        src="GardenGood-logo.svg"
      />
      <div
        class="Home__intro-text"
      >
        <h1>
          Helping gardeners plan, plant, and tend their crops for a thriving garden.
        </h1>
        <h2>
          It's time for you to garden GOOD.
        </h2>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Router component tests matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main
      class="Home"
    >
      <div
        class="Home__bg"
      />
      <img
        alt="GardenGood"
        class="Home__logo"
        src="GardenGood-logo.svg"
      />
      <div
        class="Home__intro-text"
      >
        <h1>
          Helping gardeners plan, plant, and tend their crops for a thriving garden.
        </h1>
        <h2>
          It's time for you to garden GOOD.
        </h2>
      </div>
      <p>
        <a
          href="/login"
        >
          Login
        </a>
        <a
          href="/signup"
        >
          Signup
        </a>
      </p>
    </main>
  </div>
</DocumentFragment>
`;
